### PR TITLE
fix(announcements): directive update handling of nonexistent feature flags

### LINKED ISSUE #27708: Directive Update: Handling of Nonexistent Feature Flags
### Parent Issue

#25733 

### Task

We are turning on the feature flag for announcements, but we are changing the way we activate them. Previously, by default, a feature flag was set to false if the property was not explicitly specified. However, we want to change this so that if the property is not specified, it defaults to true.

This change entails a modification in the front-end behavior that has not yet been implemented. However, I understand that Falcon has made a patch to activate it.

### Proposed Objective

Core Features

### Proposed Priority

Priority 3 - Average

### Acceptance Criteria

_No response_

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by manuelrojas**
### Proposed Changes

This pull request updates the feature flag directive to accommodate scenarios where the flag is not explicitly specified. In such cases, the directive now defaults to displaying the content.

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

File: core-web/libs/data-access/src/lib/dot-properties/dot-properties.service.ts [hunk 1]
```diff
@@ -57,13 +57,21 @@ export class DotPropertiesService {
...
      * Get the value of specific feature flag
-     *
+     * Some features we want to display in case of not found.
      * @param {FeaturedFlags} key
      * @return {*}  {Observable<boolean>}
      * @memberof DotPropertiesService
      */
-    getFeatureFlag(key: FeaturedFlags): Observable<boolean> {
-        return this.getKey(key).pipe(map((value) => value === 'true'));
+    getFeatureFlag(key: FeaturedFlags, onNotFound = false): Observable<boolean> {
```
Comment (rjvelazco) [inline core-web/libs/data-access/src/lib/dot-properties/dot-properties.service.ts:65]: Add a test case for then new param

File: core-web/apps/dotcms-ui/src/app/shared/directives/dot-show-hide-feature/dot-show-hide-feature.directive.ts [hunk 2]
```diff
@@ -63,18 +68,20 @@ export class DotShowHideFeatureDirective implements OnInit {
     ) {}
 
     ngOnInit() {
-        this.dotPropertiesService.getFeatureFlag(this._featureFlag).subscribe((isEnabled) => {
-            this.viewContainer.clear();
+        this.dotPropertiesService
+            .getFeatureFlag(this._featureFlag, this._showOnNotFound)
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/shared/directives/dot-show-hide-feature/dot-show-hide-feature.directive.ts:72]: Add a test case

**Review by rjvelazco — CHANGES_REQUESTED** [review]

File: core-web/libs/data-access/src/lib/dot-properties/dot-properties.service.ts (hunk 1 above)
Comment (zJaaal) [reply core-web/libs/data-access/src/lib/dot-properties/dot-properties.service.ts:65]: `onNotFound` sounds like a callback to me. What about `shouldAlwaysShow`?

**Review by zJaaal — CHANGES_REQUESTED** [review]

**Comment by zJaaal** [comment]
> `shouldAlwaysShow` sounds like it should be displayed even when the FF returns `false`. In this case, we want to show the feature if the feature flag is not set. @zJaaal
> 
> P.S.: This is what I understand from the issue.

Maybe we can find another name to it. My concern is that `onNotFound` sounds more like a callback than to a boolean, which can be confusing.

**Review by oidacra — APPROVED** [review]

**Review by fmontes — CHANGES_REQUESTED** [review]

**Review by hmoreras — APPROVED** [review]

**Review by rjvelazco — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

File: core-web/libs/data-access/src/lib/dot-properties/dot-properties.service.ts [hunk 3]
```diff
@@ -57,13 +57,14 @@ export class DotPropertiesService {
...
      * Get the value of specific feature flag
-     *
      * @param {FeaturedFlags} key
      * @return {*}  {Observable<boolean>}
      * @memberof DotPropertiesService
      */
     getFeatureFlag(key: FeaturedFlags): Observable<boolean> {
-        return this.getKey(key).pipe(map((value) => value === 'true'));
+        return this.getKey(key).pipe(
+            map((value) => (value === FEATURE_FLAG_NOT_FOUND ? true : value === 'true'))
```
Comment (zJaaal) [inline core-web/libs/data-access/src/lib/dot-properties/dot-properties.service.ts:66]: So this means that if we hit a FF that is not in the env we anyways will show it? 
What if a client doesn't config those FF in its env? 
Can we add test cases for that?

I am not sure if we really want to show the feature if the flag is not found all the time.

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: core-web/libs/data-access/src/lib/dot-properties/dot-properties.service.ts (hunk 3 above)
Comment (manuelrojas) [reply core-web/libs/data-access/src/lib/dot-properties/dot-properties.service.ts:66]: @zJaaal There is a unit test for this service but I case of `FEATURE_FLAG_NOT_FOUND` It needs to behave exactly as true based on the meeting had with @fmontes yesterday.

**Review by zJaaal — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

**Pull request merged**